pkg/controller/volume/attacherdetacher/attacher_detacher.go

```diff
@@ -72,29 +72,29 @@ type attacherDetacher struct {
 }
 
 func (ad *attacherDetacher) AttachVolume(
-	volumeToAttach *cache.VolumeToAttach,
+	volumeToAttach cache.VolumeToAttach,
 	actualStateOfWorld cache.ActualStateOfWorld) error {
 	attachFunc, err := ad.generateAttachVolumeFunc(volumeToAttach, actualStateOfWorld)
 	if err != nil {
 		return err
 	}
 
-	return ad.pendingOperations.Run(volumeToAttach.VolumeName, attachFunc)
+	return ad.pendingOperations.Run(string(volumeToAttach.VolumeName), attachFunc)
 }
 
 func (ad *attacherDetacher) DetachVolume(
-	volumeToDetach *cache.AttachedVolume,
+	volumeToDetach cache.AttachedVolume,
 	actualStateOfWorld cache.ActualStateOfWorld) error {
 	detachFunc, err := ad.generateDetachVolumeFunc(volumeToDetach, actualStateOfWorld)
 	if err != nil {
 		return err
 	}
 
-	return ad.pendingOperations.Run(volumeToDetach.VolumeName, detachFunc)
+	return ad.pendingOperations.Run(string(volumeToDetach.VolumeName), detachFunc)
 }
 
 func (ad *attacherDetacher) generateAttachVolumeFunc(
-	volumeToAttach *cache.VolumeToAttach,
+	volumeToAttach cache.VolumeToAttach,
 	actualStateOfWorld cache.ActualStateOfWorld) (func() error, error) {
 	// Get attacher plugin
 	attachableVolumePlugin, err := ad.volumePluginMgr.FindAttachablePluginBySpec(volumeToAttach.VolumeSpec)
```

Comment:
the line below this is collapsing an error state and a non-error state, if I'm not mistaken.